### PR TITLE
[Merged by Bors] - Use OpenSSL by default for k8-client

### PR DESCRIPTION
**Pull request by simlay**

### CODE REVIEW

File: src/k8-client/src/error.rs [hunk 1]
```diff
@@ -27,6 +27,7 @@ pub enum ClientError {
     PatchError,
     HyperError(HyperError),
     Client(StatusCode),
+    Tls(fluvio_future::openssl::TlsError),
```
Comment (sehz) [inline src/k8-client/src/error.rs:30]: This is  feature flag dependent

**Review by sehz — CHANGES_REQUESTED** [review]
Looks good.  A minor changes is needed to make sure error are feature dependent

File: src/k8-client/src/error.rs (hunk 1 above)
Comment (simlay) [reply src/k8-client/src/error.rs]: done.

**Review by sehz — APPROVED** [review]
LGTM.

**Comment by sehz** [comment]
rebase fun

**Pull request closed without merging**